lib/bundler/feature_flag.rb

```diff
@@ -64,6 +65,12 @@ def initialize(bundler_version)
       @bundler_version = Gem::Version.create(bundler_version)
     end
 
+    def github_https(&block)
+      value = Bundler.settings["github.https"]
+      value = instance_eval(&default) if value.nil? && !default.nil?
+      value
+    end
```

Comment:
Instead of having a specific reader for this feature flag, we can maybe add general support for "dot settings" and use a similar criteria to the one for translating environment variables into settings and viceversa. Something like

```diff
diff --git a/lib/bundler/feature_flag.rb b/lib/bundler/feature_flag.rb
index 83e7ff038..c39814c34 100644
--- a/lib/bundler/feature_flag.rb
+++ b/lib/bundler/feature_flag.rb
@@ -18,7 +18,7 @@ module Bundler
 
     def self.settings_method(name, key, &default)
       define_method(name) do
-        value = Bundler.settings[key]
+        value = Bundler.settings[key.gsub("__", ".")]
         value = instance_eval(&default) if value.nil? && !default.nil?
         value
       end
@@ -57,6 +57,7 @@ module Bundler
     settings_flag(:update_requires_all_flag) { bundler_2_mode? }
     settings_flag(:use_gem_version_promoter_for_major_updates) { bundler_2_mode? }
     settings_flag(:viz_command) { !bundler_2_mode? }
+    settings_flag(:github__https) { bundler_2_mode? }
 
     settings_option(:default_cli_command) { bundler_2_mode? ? :cli_help : :install }
```

Reply:
I don't want to make this something that is supported in Bundler, this PR is very much an exception to the usual rule.

Reply:
Ok @colby-swandale, if you think that's best, it's no big deal.